hivemind/dht/node.py

```diff
@@ -400,3 +520,45 @@ async def _refresh_routing_table(self, *, period: Optional[float]) -> None:
                 await self.find_nearest_nodes(refresh_id)
 
             await asyncio.sleep(max(0.0, period - (get_dht_time() - refresh_time)))
+
+
+@dataclass(init=True, repr=True, frozen=False, order=False)
+class _IntermediateResult:
```

Comment:
Maybe it's best to move the definition to the top before any classes that use it?

Reply:
I agree with the concern, but the rest of the code is structured top-down (i.e. get_many -> get_many_by_id -> _update_cache), and its the same with other modules. If you don't mind, i would stick to the standard for now, but we can change it later in a less sneaky way.

Reply:
Ok, let's create a discussion issue